internal/provider/resource_private_key.go

```diff
@@ -125,26 +166,50 @@ func CreatePrivateKey(d *schema.ResourceData, meta interface{}) error {
 		if err != nil {
 			return fmt.Errorf("error encoding key to PEM: %s", err)
 		}
+
 		keyPemBlock = &pem.Block{
 			Type:  "EC PRIVATE KEY",
 			Bytes: keyBytes,
 		}
+
+		// GOTCHA: `x/crypto/ssh` doesn't handle elliptic curve P-224
+		if k.Curve.Params().Name == "P-224" {
+			doMarshallOpenSSHKeyPemBlock = false
+		}
+	case *ed25519.PrivateKey:
+		keyBytes, err := x509.MarshalPKCS8PrivateKey(*k)
+		if err != nil {
+			return fmt.Errorf("error encoding key to PEM: %s", err)
+		}
+
+		keyPemBlock = &pem.Block{
+			Type:  "PRIVATE KEY",
+			Bytes: keyBytes,
+		}
 	default:
 		return fmt.Errorf("unsupported private key type")
 	}
-	keyPem := string(pem.EncodeToMemory(keyPemBlock))
+	d.Set("private_key_pem", string(pem.EncodeToMemory(keyPemBlock)))
 
-	d.Set("private_key_pem", keyPem)
+	// Marshall the Key in OpenSSH PEM block, if enabled
+	d.Set("private_key_openssh", "")
+	if doMarshallOpenSSHKeyPemBlock {
+		openSSHKeyPemBlock, err := openssh.MarshalPrivateKey(key, "")
+		if err != nil {
+			return err
```

Comment:
Nit: Let's wrap the error here so its clearer for practitioners about what happened

```suggestion
			return fmt.Errorf("unable to marshal private key into OpenSSH format: %w", err)
```